java/core/src/java/org/apache/orc/impl/reader/tree/TypeReader.java

```diff
@@ -24,22 +25,70 @@
 import org.apache.orc.impl.reader.StripePlanner;
 
 import java.io.IOException;
+import java.util.EnumSet;
 
 public interface TypeReader {
   void checkEncoding(OrcProto.ColumnEncoding encoding) throws IOException;
 
-  void startStripe(StripePlanner planner) throws IOException;
+  void startStripe(StripePlanner planner, ReadPhase readPhase) throws IOException;
 
-  void seek(PositionProvider[] index) throws IOException;
+  void seek(PositionProvider[] index, ReadPhase readPhase) throws IOException;
 
-  void seek(PositionProvider index) throws IOException;
+  void seek(PositionProvider index, ReadPhase readPhase) throws IOException;
 
-  void skipRows(long rows) throws IOException;
+  void skipRows(long rows, ReadPhase readPhase) throws IOException;
 
   void nextVector(ColumnVector previous,
                   boolean[] isNull,
                   int batchSize,
-                  FilterContext filterContext) throws IOException;
+                  FilterContext filterContext,
+                  ReadPhase readPhase) throws IOException;
 
   int getColumnId();
```

Comment:
not sure we need getColumnId() anymore?

Reply:
Sure. This was not added as part of this PR. I can raise a separate PR to remove this or include the removal here.

Reply:
Lets do that as part of the follow up PR -- this part of the code is not released yet (only part of main branch) so we have more flexibility